docs/source/conf.py

```diff
@@ -308,3 +311,29 @@
 
 # How to display URL addresses: 'footnote', 'no', or 'inline'.
 #texinfo_show_urls = 'footnote'
+
+
+commit_hash = os.environ.get("SOPEL_GIT_COMMIT", "")[:7]
+is_dirty = bool(os.environ.get("SOPEL_GIT_DIRTY"))
+if commit_hash:
+    github_ref = "https://github.com/sopel-irc/sopel/commit/{commit_hash}".format(commit_hash=commit_hash)
+    build_info = "(built against `{commit_hash}{dirty} <{github_ref}>`_)".format(
+        commit_hash=commit_hash[:7],
+        dirty="-dirty" if is_dirty else "",
+        github_ref=github_ref
+    )
+else:
+    build_info = "(built against an unknown commit)"
+
+if Version(__version__).is_prerelease or commit_hash:
+    tags.add("preview")
```

Comment:
Note: with the change to an environment variable, the `preview` tag is added if the user set the environment variable, even if the `__version__` is not a pre-release. It seems prudent that the output of `make docs_preview` would always set the tag

Reply:
Hm... this might sound stupid but: why do we need to check the commit hash when we can just check for the version number? It feels like an awful lot of works for something that shouldn't happen outside of the commit that bumps Sopel's version to a stable one.

So actually, I'm :-1: on the whole commit hash check.